internal/services/datafactory/data_factory_dataset_cosmosdb_mongoapi_resource_test.go

```diff
@@ -0,0 +1,169 @@
+package datafactory_test
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+	"github.com/hashicorp/terraform-plugin-testing/plancheck"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/acceptance"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/acceptance/check"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/clients"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/datafactory/parse"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+)
+
+type DatasetCosmosDbMongoDbResource struct{}
+
+func TestAccDataFactoryDatasetCosmosDbMongoDb_basic(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_data_factory_dataset_cosmosdb_mongoapi", "test")
+	r := DatasetCosmosDbMongoDbResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func TestAccDataFactoryDatasetCosmosDbMongoDb_requiresImport(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_data_factory_dataset_cosmosdb_mongoapi", "test")
+	r := DatasetCosmosDbMongoDbResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.RequiresImportErrorStep(r.requiresImport),
+	})
+}
+
+func TestAccDataFactoryDatasetCosmosDbMongoDb_complete(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_data_factory_dataset_cosmosdb_mongoapi", "test")
+	r := DatasetCosmosDbMongoDbResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.complete(data),
+		},
+		data.ImportStep(),
+	})
+}
+
+func TestAccDataFactoryDatasetCosmosDbMongoDb_update(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_data_factory_dataset_cosmosdb_mongoapi", "test")
+	r := DatasetCosmosDbMongoDbResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+		},
+		data.ImportStep(),
+		{
+			Config: r.complete(data),
+			ConfigPlanChecks: resource.ConfigPlanChecks{
+				PreApply: []plancheck.PlanCheck{
+					plancheck.ExpectResourceAction("azurerm_data_factory_dataset_cosmosdb_mongoapi.test", plancheck.ResourceActionUpdate),
+				},
+			},
+		},
+		data.ImportStep(),
```

Comment:
Can we add one more step to update it back from complete to basic?

Reply:
Added in 1179a4200a